docs/ingestion/automatic-compaction.md

```diff
@@ -0,0 +1,196 @@
+
+```sh
+curl --location --request DELETE 'http://localhost:8081/druid/coordinator/v1/config/compaction/wikipedia'
+```
+
+
+## Configure automatic compaction
+
+You can configure automatic compaction dynamically without restarting Druid.
+The automatic compaction system uses the following syntax:
+
+```json
+{
+    "dataSource": <task_datasource>,
+    "ioConfig": <IO config>,
+    "dimensionsSpec": <custom dimensionsSpec>,
+    "transformSpec": <custom transformSpec>,
+    "metricsSpec": <custom metricsSpec>,
+    "tuningConfig": <parallel indexing task tuningConfig>,
+    "granularitySpec": <compaction task granularitySpec>,
+    "skipOffsetFromLatest": <time period to avoid compaction>,
+    "taskPriority": <compaction task priority>,
+    "taskContext": <task context>
+}
+```
+
+Most fields in the auto-compaction configuration align with a typical [Druid ingestion spec](../ingestion/ingestion-spec.md).
+The following properties only apply to auto-compaction:
+* `skipOffsetFromLatest`
+* `taskPriority`
+* `taskContext`
+
+Since the automatic compaction system provides a management layer on top of manual compaction tasks,
+the auto-compaction configuration does not include task-specific properties found in a typical Druid ingestion spec.
+The following properties are automatically set by the Coordinator:
+* `type`: Set to `compact`.
+* `id`: Generated using the task type, datasource name, interval, and timestamp.
+* `context`: Set according to the user-provided `taskContext`.
+
+For more details on each of the specs in an auto-compaction configuration, see [Automatic compaction dynamic configuration](../configuration/index.md#automatic-compaction-dynamic-configuration).
+
+
+### Avoid conflicts with ingestion
+
+The Coordinator compacts segments from newest to oldest. In the auto-compaction configuration, you can set a time period, relative to the end time of the most recent segment, for segments that should not be compacted. Assign this value to `skipOffsetFromLatest`. Note that this offset is not relative to the current time but to the latest segment time. For example, if you want to skip over segments from thirty days prior to the end time of the most recent segment, assign `"skipOffsetFromLatest": "P30D"`.
+
+Compaction tasks that interfere with ingestion tasks will fail. For example, this occurs when an ingestion task needs to write data to a segment for a time interval locked for compaction. To facilitate the continuance of compaction tasks, consider one of the following strategies:
+* Set `skipOffsetFromLatest` to reduce the chance of conflicts between ingestion and compaction.
+* Increase the priority value of compaction tasks relative to ingestion tasks. In the auto-compaction configuration, set `taskPriority` to the desired priority value. See [Lock priority](../ingestion/tasks.md#lock-priority) for the priority values of different task types.
```

Comment:
We should call out that increasing the priority means that the other ingestion task will fail. If compaction job is competing with streaming ingestion, this means the streaming ingestion task will lag. This is not ideal in most cases and increasing task priority should only be used in very narrow use cases

Reply:
Yeah, I think we shouldn't advise increasing priority at all. Or atleast under a big warn sign saying "for advanced users only" or something.

Reply:
```suggestion
* Increase the priority value of compaction tasks relative to ingestion tasks. Only recommended for advanced users. This approach can cause ingestion jobs to fail or lag. To change the priority of compaction tasks, set `taskPriority` to the desired priority value in the auto-compaction configuration. For details on the priority values of different task types, see [Lock priority](../ingestion/tasks.md#lock-priority).
```